Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift

```diff
@@ -276,6 +277,37 @@ class ConversationSearchViewController: UITableViewController {
         return 4
     }
 
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        guard let view = self.tableView(tableView, viewForHeaderInSection: section) else {
+            return 0
+        }
+        return view.height()
+    }
+
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let title = self.tableView(tableView, titleForHeaderInSection: section) else {
+            return nil
+        }
+
+        let label = UILabel()
+        label.textColor = Theme.secondaryColor
+        label.text = title
+        label.font = UIFont.ows_dynamicTypeBody.ows_mediumWeight()
+        label.tag = section
+        label.sizeToFit()
+
+        let hMargin: CGFloat = 15
+        let vMargin: CGFloat = 4
+        let wrapper = UIView()
+        wrapper.backgroundColor = Theme.offBackgroundColor
+        wrapper.addSubview(label)
+        label.autoPinWidthToSuperview(withMargin: hMargin)
+        label.autoPinHeightToSuperview(withMargin: vMargin)
+        wrapper.frame = CGRect(x: 0, y: 0, width: label.width() + 2 * hMargin, height: label.height() + 2 * vMargin)
```

Comment:
Huh, we've got kind of a confusing mix of autolayout and manual sizing here, and typically the sizing logic would go in the `heightForHeaderInSection` method. 

Can we instead do all the layout with autolayout and have `heightForHeaderInSection` return `UITableViewAutomaticDimension `?

Not a deal breaker for me, it just seems like it'll be easier if we need to come back and tweak the layout later to avoid this manual sizing logic.